### PR TITLE
Rename to @metamask/inpage-provider

### PR DESCRIPTION
**Pull request by rekmarks**
- Rename to `@metamask/inpage-provider`
  - Fix description in `package.json`

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v10.17.0
```
Comment (Gudahtt) [inline .nvmrc]: Could this be moved to a separate PR? It doesn't seem related to the rename

**Review by Gudahtt — CHANGES_REQUESTED** [review]
The README has a few references to the old name as well

File: .nvmrc (hunk 1 above)
Comment (rekmarks) [reply .nvmrc]: Please no. I updated the PR title and description such that it is related

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: .nvmrc (hunk 1 above)
Comment (rekmarks) [reply .nvmrc]: For the record: I submitted
